### PR TITLE
Add 4 quick action hotkeys to game preferences

### PR DESCRIPTION
**Pull request by Chubbygummibear**
# Document the changes in your pull request

adds 4 hotkey options for quickly activating actions to the game preferences. prime a spell, activate an item like airshoes, use a power with a button press.

good luck figuring out which keys you can assign that are easy to reach and aren't already integral to this god forsaken game. Spacebar is the only one i found easy to use

Items and spells

https://user-images.githubusercontent.com/46236974/222831819-db91a50f-3f73-4923-a028-cd9046ca1a04.mp4

simple mobs

https://user-images.githubusercontent.com/46236974/222831836-4435e961-3051-4373-ad67-41261effa7ae.mp4

benos

https://user-images.githubusercontent.com/46236974/222831853-1871a26c-69a7-4c76-9aa6-e7f0e8cef409.mp4

Actual League of Legends gameplay

https://user-images.githubusercontent.com/46236974/222850681-ae8e4911-5506-447a-a084-d6060eed124e.mp4

# Wiki Documentation

Don't think any

# Changelog

:cl:  
rscadd: 4 Hotkey options for actions to the carbon tab of keybinds
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
your moms a hoe

**Comment by Mqiib** [comment]
limited to 4, seems fine to me

**Comment by MajManatee** [comment]
based as hell
People with a morbillion mouse buttons rejoice

**Comment by Chubbygummibear** [comment]
> based as hell People with a morbillion mouse buttons rejoice

not quite. byong refused to recognize m3, back button and forward button on my mouse. it only seems to be able to interpret left and right "clicks"

**Comment by Burger1998** [comment]
buster arm buff

File: code/datums/keybinding/mob_actions.dm [hunk 1]
```diff
@@ -0,0 +1,39 @@
+///Hotkeys for performing actions
+/datum/keybinding/mob/action_hotkey
+	var/action_num = 0
+	category = CATEGORY_CARBON
+	description = "Hotkey to activate an action of your mob in the corresponding slot"
+
+/datum/keybinding/mob/action_hotkey/can_use(client/user)
+	return user?.mob?.actions?.len>=action_num && ..()
```
Comment (ToasterBiome) [inline code/datums/keybinding/mob_actions.dm:8]: i hate this

File: code/datums/keybinding/mob_actions.dm [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	var/datum/action/linked_action = M.actions[action_num]
+	if (linked_action)
+		linked_action.Trigger()
+		SEND_SOUND(usr, get_sfx("terminal_type"))
+	else
+		to_chat(user, span_warning("You don't have an action to use in that slot"))
+	return TRUE
+
+/datum/keybinding/mob/action_hotkey/action_1
+	// hotkey_keys = list("Shift!")
```
Comment (ToasterBiome) [inline code/datums/keybinding/mob_actions.dm]: remove

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

File: code/datums/keybinding/mob_actions.dm (hunk 2 above)
Comment (Chubbygummibear) [reply code/datums/keybinding/mob_actions.dm]: ok

File: code/datums/keybinding/mob_actions.dm (hunk 1 above)
Comment (Chubbygummibear) [reply code/datums/keybinding/mob_actions.dm:8]: i'm open to suggestions lmao

**Comment by MajManatee** [comment]
> > based as hell People with a morbillion mouse buttons rejoice
> 
> not quite. byong refused to recognize m3, back button and forward button on my mouse. it only seems to be able to interpret left and right "clicks"

Most, if not all, mouse software's can rebind keys to anything. My mouse buttons are bound to random things like "Unikey 263" or something. 

not that im complaining, its gonna be great.

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**